Packages/Tracking/Physical Hands/Runtime/Scripts/PhysicalHandsButtonBase.cs

```diff
@@ -0,0 +1,344 @@
+        {
+            // Ensure the pressable object has a Rigidbody component
+            if (!_pressableObject.TryGetComponent<Rigidbody>(out _pressableObjectRB))
+            {
+                _pressableObjectRB = _pressableObject.AddComponent<Rigidbody>();
+                _pressableObjectRB.useGravity = false;
+            }
+
+
+
+            // Ensure the pressable object has an IgnorePhysicalHands component
+            if (!_pressableObject.TryGetComponent<IgnorePhysicalHands>(out IgnorePhysicalHands ignorePhysHands))
+            {
+                ignorePhysHands = _pressableObject.AddComponent<IgnorePhysicalHands>();
+                ignorePhysHands.DisableAllGrabbing = true;
+                ignorePhysHands.DisableAllHandCollisions = false;
+                ignorePhysHands.DisableCollisionOnChildObjects = false;
+            }
+
+            // Ensure the pressable object has a PhysicalHandsButtonHelper component
+            if (!_pressableObject.TryGetComponent<PhysicalHandsButtonHelper>(out _buttonHelper))
+            {
+                _buttonHelper = _pressableObject.AddComponent<PhysicalHandsButtonHelper>();
+            }
+
+            // Subscribe to events from the button helper
+            _buttonHelper._onHandContact += OnHandContactPO;
+            _buttonHelper._onHandContactExit += OnHandContactExitPO;
+            _buttonHelper._onHandHover += OnHandHoverPO;
+            _buttonHelper._onHandHoverExit += OnHandHoverExitPO;
+            _buttonHelper._onCollisionEnter += OnCollisionPO;
+            _buttonHelper._onCollisionExit += OnCollisionExitPO;
+        }
+
+        /// <summary>
+        /// Set up the spring joint for the button.
+        /// </summary>
+        private void SetUpSpringJoint()
+        {
+            if (!_pressableObject.TryGetComponent<ConfigurableJoint>(out _configurableJoint))
+            {
+                _configurableJoint = _pressableObject.AddComponent<ConfigurableJoint>();
+            }
+
+            // Connect the button to the parent object with a spring joint
+            
+            _configurableJoint.connectedBody = _rigidbody;
+
+            // Configure spring parameters
+            _configurableJoint.yDrive = new JointDrive
+            {
+                positionSpring = 100,
+                positionDamper = 10,
+                maximumForce = 1
+            };
+
+            // Lock and limit motion axes
+            _configurableJoint.xMotion = ConfigurableJointMotion.Locked;
+            _configurableJoint.yMotion = ConfigurableJointMotion.Limited;
+            _configurableJoint.zMotion = ConfigurableJointMotion.Locked;
+            _configurableJoint.angularXMotion = ConfigurableJointMotion.Locked;
+            _configurableJoint.angularYMotion = ConfigurableJointMotion.Locked;
+            _configurableJoint.angularZMotion = ConfigurableJointMotion.Locked;
+
+            // Adjust anchor position for button travel distance
```

Comment:
i wonder if we can make it so the start position is the furthest the button can move - currently it seems to be the middle - which feels odd, allowing the button the ability to pull away from the base